### ISSUE TITLE
VS Code times out when trying to sign in to backup

### ISSUE THREAD

**Issue by alexis-opolka** [microsoft/vscode#231198]
Does this issue occur when all extensions are disabled?: `Yes`/No

- VS Code Version:  1.94.2 (`384ff7382de624fb94dbaf6da11977bba1ecd427`)
- OS Version: `Debian 12.7.0`, linux kernel: `6.1.0-26-amd64`

Steps to Reproduce:

1. Install vscode (either via the binary or the package manager) on a fresh Debian 12 install (kernel version: `6.1.0-26-amd64` and Debian version: `12.7.0`)
2. Try to sign in to restore settings and extensions
3. See vscode timing out with the "Visual Studio Code is not responding" message

This issue seems to be related to #195800.

After having enabled the debug level on trace, here are the logs:

- when the `password-store` is set to `gnome-libsecret` <-- vscode times out:

  <details>
  <summary>
  See more
  </summary>
  
    ```txt
    2024-10-12 22:35:53.134 [trace] [File Watcher (node.js)] Request to start watching: /home/centaurus/.config/Code/User (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>),/home/centaurus/.config/Code/User/settings.json (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>)
    2024-10-12 22:35:53.134 [debug] Starting VS Code
    2024-10-12 22:35:53.134 [debug] from: /usr/share/code/resources/app
    2024-10-12 22:35:53.134 [debug] args: {"_":[],"diff":false,"merge":false,"add":false,"goto":false,"new-window":false,"reuse-window":false,"wait":false,"help":false,"list-extensions":false,"show-versions":false,"pre-release":false,"update-extensions":false,"version":false,"verbose":true,"status":false,"prof-startup":false,"no-cached-data":false,"prof-v8-extensions":false,"disable-extensions":false,"disable-lcd-text":false,"disable-gpu":false,"disable-chromium-sandbox":false,"sandbox":false,"telemetry":false,"debugRenderer":false,"enable-smoke-test-driver":false,"logExtensionHostCommunication":false,"skip-release-notes":false,"skip-welcome":false,"disable-telemetry":false,"disable-updates":false,"use-inmemory-secretstorage":false,"password-store":"gnome-libsecret","disable-workspace-trust":false,"disable-crash-reporter":false,"crash-reporter-id":"9f59b36e-59a8-4669-8e4c-ef262daec477","skip-add-to-recently-opened":false,"open-url":false,"file-write":false,"file-chmod":false,"force":false,"do-not-sync":false,"trace":false,"preserve-env":false,"force-user-env":false,"force-disable-user-env":false,"open-devtools":false,"disable-gpu-sandbox":false,"__enable-file-policy":false,"enable-coi":false,"no-proxy-server":false,"no-sandbox":false,"nolazy":false,"force-renderer-accessibility":false,"ignore-certificate-errors":false,"allow-insecure-localhost":false,"vmodule":"*/components/os_crypt/*=1","disable-dev-shm-usage":false,"profile-temp":false,"logsPath":"/home/centaurus/.config/Code/logs/20241012T223552"}
    2024-10-12 22:35:53.134 [trace] Resolving machine identifier...
    2024-10-12 22:35:53.134 [trace] Resolved machine identifier: 5a40e4366d65346552880fb26c89907a3f6f294a506c653f0b39044726b25873
    2024-10-12 22:35:53.134 [trace] Main->SharedProcess#connect
    2024-10-12 22:35:53.134 [trace] [File Watcher (node.js)] Started watching: '/home/centaurus/.config/Code/User'
    2024-10-12 22:35:53.134 [trace] [File Watcher (node.js)] ignoring a path for watching who's stat info failed to resolve: /home/centaurus/.config/Code/User/settings.json (error: Error: ENOENT: no such file or directory, stat '/home/centaurus/.config/Code/User/settings.json')
    2024-10-12 22:35:53.134 [trace] [File Watcher (node.js)] starting fs.watchFile() on /home/centaurus/.config/Code/User/settings.json (correlationId: undefined)
    2024-10-12 22:35:53.134 [trace] StorageMainService: creating application storage
    2024-10-12 22:35:53.134 [trace] ElectronURLListener: waiting for window to be ready to handle URLs...
    2024-10-12 22:35:53.134 [trace] lifecycle (main): phase changed (value: 2)
    2024-10-12 22:35:53.134 [trace] windowsManager#open
    2024-10-12 22:35:53.134 [trace] windowsManager#open pathsToOpen [{"backupPath":"/home/centaurus/.config/Code/Backups/1728763710866"}]
    2024-10-12 22:35:53.134 [trace] windowsManager#doOpenEmpty {"restore":true,"forceNewWindow":true}
    2024-10-12 22:35:53.134 [trace] IPC Object URL: Registered new channel vscode:e101a3a6-aef5-4bc4-b94d-5fdd9c51f0e2.
    2024-10-12 22:35:53.134 [trace] window#validateWindowState: validating window state on 1 display(s) {"mode":1,"x":448,"y":156,"width":1024,"height":768}
    2024-10-12 22:35:53.134 [trace] window#validateWindowState: single monitor working area {"x":0,"y":32,"width":1920,"height":1048}
    2024-10-12 22:35:53.134 [trace] window#ctor: using window state {"mode":1,"x":448,"y":156,"width":1024,"height":768}
    2024-10-12 22:35:53.134 [trace] window#load: attempt to load window (id: 1)
    2024-10-12 22:35:53.134 [trace] windowsManager#open used window count 1 (workspacesToOpen: 0, foldersToOpen: 0, emptyToRestore: 1, openOneEmptyWindow: false)
    2024-10-12 22:35:53.134 [trace] lifecycle (main): phase changed (value: 3)
    2024-10-12 22:35:53.134 [trace] resolveShellEnv(): skipped (VSCODE_CLI is set)
    2024-10-12 22:35:53.134 [info] update#setState idle
    2024-10-12 22:35:53.221 [trace] resolveShellEnv(): skipped (VSCODE_CLI is set)
    2024-10-12 22:35:53.590 [trace] StorageMainService: creating workspace storage (1728763710866)
    2024-10-12 22:35:53.740 [trace] [File Watcher (node.js)] Request to stop watching: /home/centaurus/.config/Code/User/settings.json (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>)
    2024-10-12 22:35:53.741 [trace] [File Watcher (node.js)] stopping file watcher (/home/centaurus/.config/Code/User/settings.json (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>))
    2024-10-12 22:35:53.878 [trace] window#load: window reported ready (id: 1)
    2024-10-12 22:35:53.879 [trace] ElectronURLListener#flush(): flushing URLs
    2024-10-12 22:35:53.886 [trace] [SharedProcess] onWindowConnection for: vscode:createSharedProcessChannelConnectionResult
    2024-10-12 22:35:53.888 [trace] [UtilityProcess type: shared-process, pid: <none>]: creating new...
    2024-10-12 22:35:53.898 [trace] [UtilityProcess type: shared-process, pid: <none>]: payload sent via postMessage()
    2024-10-12 22:35:53.900 [trace] [UtilityProcessWorker]: createWorker(window: 1, moduleId: vs/platform/files/node/watcher/watcherMain)
    2024-10-12 22:35:53.902 [trace] [UtilityProcess id: 1, type: fileWatcher, pid: <none>]: creating new...
    2024-10-12 22:35:53.917 [trace] [UtilityProcess type: shared-process, pid: 16538]: successfully created
    2024-10-12 22:35:53.918 [trace] [UtilityProcess id: 1, type: fileWatcher, pid: 16539]: successfully created
    2024-10-12 22:35:53.933 [trace] menubarService#updateMenubar 1
    2024-10-12 22:35:54.096 [trace] [UtilityProcess id: 1, type: extensionHost, pid: <none>]: creating new...
    2024-10-12 22:35:54.129 [trace] [UtilityProcess id: 1, type: extensionHost, pid: 16559]: successfully created
    2024-10-12 22:35:54.252 [trace] resolveShellEnv(): skipped (VSCODE_CLI is set)
    2024-10-12 22:35:54.377 [debug] Terminal profile not validated zsh ["zsh"]
    2024-10-12 22:35:54.377 [debug] Terminal profile not validated fish ["fish"]
    2024-10-12 22:35:54.378 [debug] Terminal profile not validated tmux ["tmux"]
    2024-10-12 22:35:54.384 [debug] Terminal profile not validated pwsh ["pwsh"]
    2024-10-12 22:35:54.415 [trace] [SharedProcess] IPC ready
    2024-10-12 22:35:54.416 [trace] Main->SharedProcess#connect: connection established
    2024-10-12 22:35:54.443 [trace] [File Watcher (node.js)] Request to start watching: /home/centaurus/.config/Code/User (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>)
    2024-10-12 22:35:54.448 [trace] [File Watcher (node.js)] Request to start watching: /home/centaurus/.config/Code/User/settings.json (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>)
    2024-10-12 22:35:54.452 [trace] [File Watcher (node.js)] Started watching: '/home/centaurus/.config/Code/User'
    2024-10-12 22:35:54.453 [trace] [File Watcher (node.js)] ignoring a path for watching who's stat info failed to resolve: /home/centaurus/.config/Code/User/settings.json (error: Error: ENOENT: no such file or directory, stat '/home/centaurus/.config/Code/User/settings.json')
    2024-10-12 22:35:54.453 [trace] [File Watcher (node.js)] starting fs.watchFile() on /home/centaurus/.config/Code/User/settings.json (correlationId: undefined)
    2024-10-12 22:35:54.480 [trace] [SharedProcess] Overall ready
    2024-10-12 22:35:54.547 [trace] [File Watcher (node.js)] Request to start watching: /home/centaurus/.vscode/extensions (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>),/home/centaurus/.vscode/extensions/extensions.json (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>)
    2024-10-12 22:35:54.547 [trace] [File Watcher (node.js)] Request to stop watching: /home/centaurus/.config/Code/User/settings.json (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>)
    2024-10-12 22:35:54.547 [trace] [File Watcher (node.js)] stopping file watcher (/home/centaurus/.config/Code/User/settings.json (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>))
    2024-10-12 22:35:54.549 [trace] [File Watcher (node.js)] Started watching: '/home/centaurus/.vscode/extensions'
    2024-10-12 22:35:54.549 [trace] [File Watcher (node.js)] Started watching: '/home/centaurus/.vscode/extensions/extensions.json'
    2024-10-12 22:35:54.784 [trace] resolveShellEnv(): skipped (VSCODE_CLI is set)
    2024-10-12 22:35:54.786 [trace] [network] #1: https://update.code.visualstudio.com/api/update/linux-x64/stable/384ff7382de624fb94dbaf6da11977bba1ecd427 - begin undefined {}
    2024-10-12 22:35:54.931 [trace] [network] #1: https://update.code.visualstudio.com/api/update/linux-x64/stable/384ff7382de624fb94dbaf6da11977bba1ecd427 - end undefined 204 {"access-control-allow-origin":"*","date":"Sat, 12 Oct 2024 20:35:54 GMT","request-context":"appId=cid-v1:8109e553-6232-492a-afca-b78cea9b431c","strict-transport-security":"max-age=31536000; includeSubDomains","x-azure-ref":"20241012T203554Z-er1f76b5c6c6wzg8cbnfz7rdy000000003r000000000126b","x-cache":"CONFIG_NOCACHE","x-content-type-options":"nosniff","x-powered-by":"Express","x-source-commit":"08c40a630cbc27e6dec48081ba4d32c45f685ee1"}
    2024-10-12 22:35:55.260 [trace] menubarService#updateMenubar 1
    2024-10-12 22:35:55.576 [trace] lifecycle (main): phase changed (value: 4)
    2024-10-12 22:35:55.861 [trace] menubarService#updateMenubar 1
    2024-10-12 22:35:56.776 [trace] menubarService#updateMenubar 1
    2024-10-12 22:35:57.437 [debug] ProxyResolver#loadSystemCertificates count 140
    2024-10-12 22:35:57.447 [debug] ProxyResolver#loadSystemCertificates count filtered 137
    2024-10-12 22:36:01.511 [trace] menubarService#updateMenubar 1
    2024-10-12 22:36:07.657 [trace] [EncryptionMainService] Checking if encryption is available...
    ```
  
</details>

- when the `password-store` is set to `basic` <-- vscode doesn't time out:

  <details>
  <summary>
  See more
  </summary>

  ```log
  2024-10-12 22:41:56.769 [trace] [File Watcher (node.js)] Request to start watching: /home/centaurus/.config/Code/User (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>),/home/centaurus/.config/Code/User/settings.json (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>)
  2024-10-12 22:41:56.769 [debug] Starting VS Code
  2024-10-12 22:41:56.769 [debug] from: /usr/share/code/resources/app
  2024-10-12 22:41:56.769 [debug] args: {"_":[],"diff":false,"merge":false,"add":false,"goto":false,"new-window":false,"reuse-window":false,"wait":false,"help":false,"list-extensions":false,"show-versions":false,"pre-release":false,"update-extensions":false,"version":false,"verbose":true,"status":false,"prof-startup":false,"no-cached-data":false,"prof-v8-extensions":false,"disable-extensions":false,"disable-lcd-text":false,"disable-gpu":false,"disable-chromium-sandbox":false,"sandbox":false,"telemetry":false,"debugRenderer":false,"enable-smoke-test-driver":false,"logExtensionHostCommunication":false,"skip-release-notes":false,"skip-welcome":false,"disable-telemetry":false,"disable-updates":false,"use-inmemory-secretstorage":false,"password-store":"basic","disable-workspace-trust":false,"disable-crash-reporter":false,"crash-reporter-id":"9f59b36e-59a8-4669-8e4c-ef262daec477","skip-add-to-recently-opened":false,"open-url":false,"file-write":false,"file-chmod":false,"force":false,"do-not-sync":false,"trace":false,"preserve-env":false,"force-user-env":false,"force-disable-user-env":false,"open-devtools":false,"disable-gpu-sandbox":false,"__enable-file-policy":false,"enable-coi":false,"no-proxy-server":false,"no-sandbox":false,"nolazy":false,"force-renderer-accessibility":false,"ignore-certificate-errors":false,"allow-insecure-localhost":false,"vmodule":"*/components/os_crypt/*=1","disable-dev-shm-usage":false,"profile-temp":false,"logsPath":"/home/centaurus/.config/Code/logs/20241012T224156"}
  2024-10-12 22:41:56.769 [trace] Resolving machine identifier...
  2024-10-12 22:41:56.769 [trace] Resolved machine identifier: 5a40e4366d65346552880fb26c89907a3f6f294a506c653f0b39044726b25873
  2024-10-12 22:41:56.769 [trace] Main->SharedProcess#connect
  2024-10-12 22:41:56.769 [trace] [File Watcher (node.js)] Started watching: '/home/centaurus/.config/Code/User'
  2024-10-12 22:41:56.769 [trace] [File Watcher (node.js)] ignoring a path for watching who's stat info failed to resolve: /home/centaurus/.config/Code/User/settings.json (error: Error: ENOENT: no such file or directory, stat '/home/centaurus/.config/Code/User/settings.json')
  2024-10-12 22:41:56.769 [trace] [File Watcher (node.js)] starting fs.watchFile() on /home/centaurus/.config/Code/User/settings.json (correlationId: undefined)
  2024-10-12 22:41:56.769 [trace] [EncryptionMainService] setting usePlainTextEncryption to true...
  2024-10-12 22:41:56.769 [trace] [EncryptionMainService] set usePlainTextEncryption to true
  2024-10-12 22:41:56.769 [trace] StorageMainService: creating application storage
  2024-10-12 22:41:56.769 [trace] ElectronURLListener: waiting for window to be ready to handle URLs...
  2024-10-12 22:41:56.769 [trace] lifecycle (main): phase changed (value: 2)
  2024-10-12 22:41:56.769 [trace] windowsManager#open
  2024-10-12 22:41:56.769 [trace] windowsManager#open pathsToOpen [{"backupPath":"/home/centaurus/.config/Code/Backups/1728763710866"}]
  2024-10-12 22:41:56.769 [trace] windowsManager#doOpenEmpty {"restore":true,"forceNewWindow":true}
  2024-10-12 22:41:56.769 [trace] IPC Object URL: Registered new channel vscode:7c26bd12-4c7c-4927-9536-a28d3dfd0fe2.
  2024-10-12 22:41:56.769 [trace] window#validateWindowState: validating window state on 1 display(s) {"mode":1,"x":448,"y":156,"width":1024,"height":768}
  2024-10-12 22:41:56.769 [trace] window#validateWindowState: single monitor working area {"x":0,"y":32,"width":1920,"height":1048}
  2024-10-12 22:41:56.769 [trace] window#ctor: using window state {"mode":1,"x":448,"y":156,"width":1024,"height":768}
  2024-10-12 22:41:56.769 [trace] window#load: attempt to load window (id: 1)
  2024-10-12 22:41:56.769 [trace] windowsManager#open used window count 1 (workspacesToOpen: 0, foldersToOpen: 0, emptyToRestore: 1, openOneEmptyWindow: false)
  2024-10-12 22:41:56.769 [trace] lifecycle (main): phase changed (value: 3)
  2024-10-12 22:41:56.769 [trace] resolveShellEnv(): skipped (VSCODE_CLI is set)
  2024-10-12 22:41:56.769 [info] update#setState idle
  2024-10-12 22:41:56.857 [trace] resolveShellEnv(): skipped (VSCODE_CLI is set)
  2024-10-12 22:41:57.114 [trace] StorageMainService: creating workspace storage (1728763710866)
  2024-10-12 22:41:57.331 [trace] window#load: window reported ready (id: 1)
  2024-10-12 22:41:57.331 [trace] ElectronURLListener#flush(): flushing URLs
  2024-10-12 22:41:57.337 [trace] [SharedProcess] onWindowConnection for: vscode:createSharedProcessChannelConnectionResult
  2024-10-12 22:41:57.339 [trace] [UtilityProcess type: shared-process, pid: <none>]: creating new...
  2024-10-12 22:41:57.348 [trace] [UtilityProcess type: shared-process, pid: <none>]: payload sent via postMessage()
  2024-10-12 22:41:57.350 [trace] [UtilityProcessWorker]: createWorker(window: 1, moduleId: vs/platform/files/node/watcher/watcherMain)
  2024-10-12 22:41:57.351 [trace] [UtilityProcess id: 1, type: fileWatcher, pid: <none>]: creating new...
  2024-10-12 22:41:57.365 [trace] [UtilityProcess type: shared-process, pid: 16882]: successfully created
  2024-10-12 22:41:57.366 [trace] [UtilityProcess id: 1, type: fileWatcher, pid: 16883]: successfully created
  2024-10-12 22:41:57.371 [trace] menubarService#updateMenubar 1
  2024-10-12 22:41:57.412 [trace] [File Watcher (node.js)] Request to stop watching: /home/centaurus/.config/Code/User/settings.json (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>)
  2024-10-12 22:41:57.412 [trace] [File Watcher (node.js)] stopping file watcher (/home/centaurus/.config/Code/User/settings.json (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>))
  2024-10-12 22:41:57.442 [trace] [UtilityProcess id: 1, type: extensionHost, pid: <none>]: creating new...
  2024-10-12 22:41:57.452 [trace] [UtilityProcess id: 1, type: extensionHost, pid: 16903]: successfully created
  2024-10-12 22:41:57.616 [trace] resolveShellEnv(): skipped (VSCODE_CLI is set)
  2024-10-12 22:41:57.711 [debug] Terminal profile not validated zsh ["zsh"]
  2024-10-12 22:41:57.711 [debug] Terminal profile not validated fish ["fish"]
  2024-10-12 22:41:57.712 [debug] Terminal profile not validated tmux ["tmux"]
  2024-10-12 22:41:57.712 [debug] Terminal profile not validated pwsh ["pwsh"]
  2024-10-12 22:41:57.768 [trace] menubarService#updateMenubar 1
  2024-10-12 22:41:57.809 [trace] [SharedProcess] IPC ready
  2024-10-12 22:41:57.810 [trace] Main->SharedProcess#connect: connection established
  2024-10-12 22:41:57.827 [trace] [File Watcher (node.js)] Request to start watching: /home/centaurus/.config/Code/User (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>),/home/centaurus/.config/Code/User/settings.json (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>)
  2024-10-12 22:41:57.829 [trace] [File Watcher (node.js)] Started watching: '/home/centaurus/.config/Code/User'
  2024-10-12 22:41:57.829 [trace] [File Watcher (node.js)] ignoring a path for watching who's stat info failed to resolve: /home/centaurus/.config/Code/User/settings.json (error: Error: ENOENT: no such file or directory, stat '/home/centaurus/.config/Code/User/settings.json')
  2024-10-12 22:41:57.830 [trace] [File Watcher (node.js)] starting fs.watchFile() on /home/centaurus/.config/Code/User/settings.json (correlationId: undefined)
  2024-10-12 22:41:57.862 [trace] [SharedProcess] Overall ready
  2024-10-12 22:41:57.928 [trace] [File Watcher (node.js)] Request to start watching: /home/centaurus/.vscode/extensions (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>),/home/centaurus/.vscode/extensions/extensions.json (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>)
  2024-10-12 22:41:57.929 [trace] [File Watcher (node.js)] Request to stop watching: /home/centaurus/.config/Code/User/settings.json (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>)
  2024-10-12 22:41:57.929 [trace] [File Watcher (node.js)] stopping file watcher (/home/centaurus/.config/Code/User/settings.json (excludes: <none>, includes: <all>, filter: <none>, correlationId: <none>))
  2024-10-12 22:41:57.930 [trace] [File Watcher (node.js)] Started watching: '/home/centaurus/.vscode/extensions'
  2024-10-12 22:41:57.930 [trace] [File Watcher (node.js)] Started watching: '/home/centaurus/.vscode/extensions/extensions.json'
  2024-10-12 22:41:58.067 [trace] resolveShellEnv(): skipped (VSCODE_CLI is set)
  2024-10-12 22:41:58.068 [trace] [network] #1: https://update.code.visualstudio.com/api/update/linux-x64/stable/384ff7382de624fb94dbaf6da11977bba1ecd427 - begin undefined {}
  2024-10-12 22:41:58.231 [trace] [network] #1: https://update.code.visualstudio.com/api/update/linux-x64/stable/384ff7382de624fb94dbaf6da11977bba1ecd427 - end undefined 204 {"access-control-allow-origin":"*","date":"Sat, 12 Oct 2024 20:41:58 GMT","request-context":"appId=cid-v1:8109e553-6232-492a-afca-b78cea9b431c","strict-transport-security":"max-age=31536000; includeSubDomains","x-azure-ref":"20241012T204158Z-er1f76b5c6cxzz8t4hppzw3s8800000003z0000000005ak9","x-cache":"CONFIG_NOCACHE","x-content-type-options":"nosniff","x-powered-by":"Express","x-source-commit":"08c40a630cbc27e6dec48081ba4d32c45f685ee1"}
  2024-10-12 22:41:58.776 [trace] menubarService#updateMenubar 1
  2024-10-12 22:41:59.215 [trace] lifecycle (main): phase changed (value: 4)
  2024-10-12 22:42:00.139 [trace] menubarService#updateMenubar 1
  2024-10-12 22:42:00.431 [debug] ProxyResolver#loadSystemCertificates count 140
  2024-10-12 22:42:00.435 [debug] ProxyResolver#loadSystemCertificates count filtered 137
  ```
  </details>

**Comment by alexis-opolka** [comment]
After a reboot, it seems to work fine now.

**Comment by TylerLeonhardt** [comment]
I'll mark this as a duplicate of #195800 since a reboot also got you unblocked, but I think we should look into this further - it's likely an electron/chromium issue.
